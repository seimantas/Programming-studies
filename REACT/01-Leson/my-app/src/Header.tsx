export const MyHeader = () => {
  return (
    <header>
      <h2 className="logo">LOGO</h2>
      <div id="meniu">
        <p>home</p>
        <p>Product</p>
        <p>Company</p>
        <p>Contact</p>
      </div>

      <h1>Header Image</h1>
    </header>
  );
};
