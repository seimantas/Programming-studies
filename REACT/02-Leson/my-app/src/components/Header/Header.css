.header-wrapper {
  padding: 10px;
  height: 200px;
  background: url("https://images.unsplash.com/photo-1464822759023-fed622ff2c3b?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=3270&q=80");
  background-position: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

h1,
h3 {
  margin: 0;
}

h1,
h3 {
  font-weight: 300;
  color: white;
}
