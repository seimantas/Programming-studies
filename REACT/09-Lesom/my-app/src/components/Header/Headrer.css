.header {
  margin-right: 50px;
  margin-left: 50px;
  border-bottom: solid 1px black;
  display: flex;
  justify-content: space-around;
}

.product-button {
  margin-top: 135px;
  height: 30px;
}
