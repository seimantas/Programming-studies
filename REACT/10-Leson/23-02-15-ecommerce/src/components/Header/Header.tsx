import { StyledLink } from "../../styles/StyledLink";
import { HeaderStyled } from "../../styles/HeaderStyled";

export const Header = () => {
  return (
    <HeaderStyled>
      <StyledLink to="/">Home</StyledLink>
      {/* <StyledLink to="/cart">Cart {cartCount}</StyledLink> */}
    </HeaderStyled>
  );
};
