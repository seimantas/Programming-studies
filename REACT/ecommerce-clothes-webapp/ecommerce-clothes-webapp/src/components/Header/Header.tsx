import { Box, Grid, Typography } from "@mui/material";
import { FC, useContext } from "react";
import { Link, useLocation } from "react-router-dom";
import { ProductsContext } from "../ProductsContext";

export const Header: FC = () => {
  const { cartProducts } = useContext(ProductsContext);
  const { pathname } = useLocation();
  const isOnCartLink = pathname.includes("/cart");

  let cartAmount = 0;

  cartProducts.forEach((product) => {
    cartAmount += product.amount;
  });

  return (
    <Box
      component="header"
      textAlign="center"
      margin="0 auto"
      // width="100"
      borderBottom="2px solid black"
    >
      <Typography variant="h3" padding={2} fontWeight="600" fontSize="44px">
        {isOnCartLink ? "CART" : "PRODUCTS"}
      </Typography>

      {/* <div role="products"></div> */}

      {/* TODO heading Products pagal link, sutvarkyti warning */}

      <Grid
        role="navigation"
        container
        textAlign="center"
        mb={2}
        sx={{
          "& a": {
            color: "purple",
            textDecoration: "none",

            ":hover": { color: "lightgreen" },
          },
        }}
      >
        <Grid item xs={12} sm={6}>
          <Link to="/">
            <Typography aria-label="home link" fontSize="32px">
              Home
            </Typography>
          </Link>
        </Grid>

        <Grid item xs={12} sm={6}>
          <Link to="/cart">
            <Typography aria-label="cart link" fontSize="32px">
              Cart ({cartAmount})
            </Typography>
          </Link>
        </Grid>
      </Grid>
    </Box>
  );
};
