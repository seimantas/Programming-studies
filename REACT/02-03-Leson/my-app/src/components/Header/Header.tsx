import "./Header.css";

export const Header = ({ title, one, two, header, children }: any) => {
  return (
    <header className={header}>
      <div className="center-content">
        <h1>{title}</h1>
        <h3>{one}</h3>
        <h3>{two}</h3>
        {children}
      </div>
    </header>
  );
};
