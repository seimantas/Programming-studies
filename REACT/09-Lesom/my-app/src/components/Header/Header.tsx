import { Link } from "react-router-dom";
import "./Headrer.css";
import { CartLogo } from "../CartLogo/CartLogo";
import { ShopLogo } from "../ShopLogo/ShopLogo";

export const Header = () => {
  return (
    <div className="header">
      <ShopLogo />
      <Link to="/">
        <h2 className="product-button">Product List</h2>
      </Link>
      <Link to="/cart">
        <CartLogo />
      </Link>
    </div>
  );
};
