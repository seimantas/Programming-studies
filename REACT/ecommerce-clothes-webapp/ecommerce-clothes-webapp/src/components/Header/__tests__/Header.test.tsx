import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Header } from "../Header";
import renderer from "react-test-renderer";

describe("Header", () => {
  //   it("should render links", () => {
  //     render(
  //       <BrowserRouter>
  //         <Header />
  //       </BrowserRouter>
  //     );

  //     expect(screen.getByLabelText("home link")).toBeVisible();
  //     expect(screen.getByLabelText("cart link")).toBeVisible();
  //   });

  it("matches snapshot", () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );

    expect(tree).toMatchSnapshot();
  });

  it.each(["home link", "cart link"])("should render %s", (link) => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );

    expect(screen.getByLabelText(link)).toBeVisible();
  });
});
