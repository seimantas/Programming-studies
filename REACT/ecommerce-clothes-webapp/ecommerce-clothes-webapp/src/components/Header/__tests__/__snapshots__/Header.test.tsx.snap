// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<header
  className="MuiBox-root css-17mas4t"
>
  <h3
    className="MuiTypography-root MuiTypography-h3 css-93metl-MuiTypography-root"
  >
    PRODUCTS
  </h3>
  <div
    className="MuiGrid-root MuiGrid-container css-16gs291-MuiGrid-root"
    role="navigation"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <p
          aria-label="home link"
          className="MuiTypography-root MuiTypography-body1 css-fldieq-MuiTypography-root"
        >
          Home
        </p>
      </a>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
    >
      <a
        href="/cart"
        onClick={[Function]}
      >
        <p
          aria-label="cart link"
          className="MuiTypography-root MuiTypography-body1 css-fldieq-MuiTypography-root"
        >
          Cart (
          0
          )
        </p>
      </a>
    </div>
  </div>
</header>
`;
